systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.upgrade;
+
+import io.strimzi.systemtest.AbstractST;
+import io.strimzi.test.TestUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.params.provider.Arguments;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+public class AbstractUpgradeST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(AbstractUpgradeST.class);
+
+    protected static JsonArray readUpgradeJson() {
+        try (InputStream fis = new FileInputStream(TestUtils.USER_PATH + "/src/main/resources/StrimziUpgradeST.json")) {
+            JsonReader reader = Json.createReader(fis);
+            return reader.readArray();
+        } catch (IOException e) {
+            e.printStackTrace();
+            throw new RuntimeException(TestUtils.USER_PATH + "/src/main/resources/StrimziUpgradeST.json" + " file was not found.");
+        }
+    }
+
+    /**
+     * List cluster operator versions which supports specific kafka version. It uses StrimziUpgradeST.json file to parse
+     * the 'toVersion' and 'proceduresAfter' JsonObjects.
+     * example:
+     *      2.6.0->[HEAD]
+     *      2.3.1->[0.15.0]
+     *      2.4.0->[0.16.2, 0.17.0]
+     *      2.5.0->[0.18.0, 0.19.0]
+     *      2.2.1->[0.12.1]
+     *      2.3.0->[0.13.0, 0.14.0]
+     * @return map key -> kafka version | value -> list of cluster operator versions
+     */
+    protected static Map<String, List<String>> getMapKafkaVersionsWithSupportedClusterOperatorVersions() {
```

Comment:
I think this method is a little bit complicated. You are building some map and based on it you are doing upgrade of Kafka and log message format. Why don't you use data from JSON directly?